### DISCUSSION TITLE
Image uploads in HEIC format

### DISCUSSION THREAD

**Discussion by VarenSwaab** [ExpressionEngine/ExpressionEngine#3000] [Feature Requests]
Uploading an image in HEIC format results in an error using EE's channel form file Fieldtype.

Would love to see support for uploading HEIC and other modern image formats.

Note that this seems to work when uploading an image directly from an iPhone because the phone must convert the image to jpg or some other compatible format but when uploading this image from a directory it results in an error.

**Comment by Sungodragon** [comment]
Is it supported in any browser?
https://caniuse.com/#search=heif

**Comment by VarenSwaab** [comment]
In this case I don't think it's a function of the browser. Safari does support it and the problem is when EE gets the file it rejects it beaus it's not a standard recognized format such as .png, .jpg, etc. Perhaps my understanding is incorrect but since it's well supported by current browsers and certainly by browsers native to operating systems that support the format, then the issue seems to be related to how EE handles these types of formats.

**Comment by derekjones** [comment]
HEIC isn't supported by common server libs either, so supporting it in PHP in general is something of a non-starter at the moment. Can I ask how you are ending up with HEIC files in a folder that you wish to upload as a content author? You correctly identify that when sharing as an image to any app that will process them, iOS automatically converts to a JPG. In my personal experience, this is always seamless.

**Comment by VarenSwaab** [comment]
Thanks for the info Derek. To answer your question, I had a user who was having trouble uploading an image using channel:form. After some poking around we discovered the format was .HEIC and with further investigation I discovered that my iPhone camera is set to .HEIC as the default image format. After some testing, I received the same error as my user. If Apple has made this the default format for the next version of iOS then this is a problem that will happen more often. As always, thanks for your quick responses.

**Comment by VarenSwaab** [comment]
Actually, to be more clear, I AirDrop'ed the file from my phone to my computer and did the upload from desktop Safari. The format my desktop received was HEIC and not jpg.

**Comment by derekjones** [comment]
Ah gotcha. When you share between compatible devices, iOS doesn't auto-convert for you. This seems like a real edge case, particularly since you'd then have to define what behavior you want ExpressionEngine to do with an HEIC. But the underlying issue still persists at this time that server libs that PHP relies on for image processing still aren't readily available. I'll leave this issue open, however, as I expect that to change at some point in the future, along with browser compatibility, which will make a solution possible.

A couple solutions for you that don't require changing the iPhone's storage default (as I'd recommend keeping the superior HEIC format):

1. Use the native Photos app and drag and drop to Finder, which will give you a jpeg.
2. If Airdropping is unavoidable, open the image in Preview first and save as a jpeg.

**Comment by VarenSwaab** [comment]
Those are great tips. Thanks Derek.

**Comment by Sungodragon** [comment]
You can reopen the issue for tracking the enhancement in the future.

**Comment by jcogs-design** [comment]
> The underlying issue still persists at this time that server libs that PHP relies on for image processing still aren't readily available.

This is no longer true. With its [1.3.7](https://jcogs.net/documentation/jcogs_img/jcogs_img-changelog#jcogs-image-changelog-1-3-7) release JCOGS Image [supports the use of HEIC files as source images](https://jcogs.net/documentation/jcogs_img/jcogs_img-advanced-topics#jcogs-image-advanced-topics-image-formats) using an [open source php library](https://github.com/MaestroError/php-heic-to-jpg).

Accordingly it would be helpful if EE could evolve to at least provide limited support for HEIC image files.

One option would be to simply allow their upload to "Image" file locations (perhaps by editing the allowed MIME types) - echoing perhaps the way that EE now is aware of SVG files but does not provide support for their manipulation; doing this would allow users to add their own manipulation capability via add-ons if required. 

A second option would be to incorporate a similar ability to optionally convert HEIC to JPG for image manipulation purposes as part of the file upload process.
